### PR TITLE
Add battery support for contingency element deserializer

### PR DESCRIPTION
**Pull request by EtienneLt**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
no

**What kind of change does this PR introduce?**
bug fix

**What is the current behavior?**
the contingency element deserializer does not deserialize Battery contingencies

**What is the new behavior (if this is a feature change)?**
add it

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

**Other information**:

### CODE REVIEW

File: contingency/contingency-api/src/main/java/com/powsybl/contingency/json/ContingencyElementDeserializer.java [hunk 1]
```diff
@@ -58,6 +58,7 @@ public ContingencyElement deserialize(JsonParser parser, DeserializationContext
                 case LOAD -> new LoadContingency(id);
                 case BUS -> new BusContingency(id);
                 case TIE_LINE -> new TieLineContingency(id, voltageLevelId);
+                case BATTERY -> new BatteryContingency(id);
```
Comment (olperr1) [inline contingency/contingency-api/src/main/java/com/powsybl/contingency/json/ContingencyElementDeserializer.java:61]: Seen with @EtienneLt: Switch contingencies are not fully supported.  
This will be the object of another PR.

**Review by olperr1 — APPROVED** [review]

**Review by So-Fras — APPROVED** [review]
